testng-core/src/test/kotlin/org/testng/internal/ClassHelperTest.kt

```diff
@@ -0,0 +1,143 @@
+package org.testng.internal
+
+import org.assertj.core.api.Assertions.assertThat
+
+import java.net.URL
+import java.net.URLClassLoader
+import java.util.UUID
+import org.testng.annotations.DataProvider
+import org.testng.annotations.Test
+import org.testng.internal.samples.classhelper.issue1339.BabyPandaSample
+import org.testng.internal.samples.classhelper.issue1339.LittlePandaSample
+import org.testng.internal.samples.classhelper.issue1456.TestClassSample
+import org.testng.internal.samples.classhelper.misamples.AbstractMovesSample
+import org.testng.internal.samples.classhelper.misamples.BatmanSample
+import org.testng.internal.samples.classhelper.misamples.JohnTravoltaMovesSample
+import org.testng.internal.samples.classhelper.misamples.MickJaggerSample
+import org.testng.xml.XmlClass
+import org.testng.xml.XmlSuite
+import org.testng.xml.XmlTest
+
+class ClassHelperTest {
+
+    @Test
+    fun testGetAvailableMethods() {
+        runTest(getExpected(), LittlePandaSample::class.java)
+    }
+
+    @Test
+    fun testGetAvailableMethodsWhenOverridingIsInvolved() {
+        val expected = getExpected("equals", "hashCode", "toString")
+        runTest(expected, BabyPandaSample::class.java)
+    }
+
+    @Test
+    fun testFindClassInSameTest() {
+        runTest(TestClassSample::class.java, 1, TestClassSample::class.java)
+    }
+
+    @Test
+    fun testFindClassesInSameTest() {
+        runTest(
+            TestClassSample::class.java, 2, TestClassSample::class.java,
+            BabyPandaSample::class.java
+        )
+    }
+
+    @Test
+    fun testNoClassDefFoundError() {
+        val urlClassLoader = object : URLClassLoader(emptyArray<URL>()) {
+            override fun loadClass(className: String?): Class<*> {
+                throw NoClassDefFoundError()
+            }
+        }
+        ClassHelper.addClassLoader(urlClassLoader)
+        val fakeClassName = UUID.randomUUID().toString()
+        assertThat(ClassHelper.forName(fakeClassName))
+            .withFailMessage("The result should be null; no exception should be thrown.")
+            .isNull()
+    }
+
+    @Test(dataProvider = "data")
+    fun testWithDefaultMethodsBeingOverridden(
+        cls: Class<*>, expectedCount: Int, vararg expected: String
+    ) {
+        val methods = ClassHelper.getAvailableMethodsExcludingDefaults(cls)
```

Comment:
`cls.getAvailableMethodsExcludingDefaults()`

Reply:
Same doubt here as well. Why do we need to define extension methods for these scenarios ? We are testing the `ClassHelper.getAvailableMethodsExcludingDefaults()`. So why should we make that as an extension method ? I am not sure i understand this ask.